server/src/main/java/org/elasticsearch/common/logging/LoggingDataProvider.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.logging;
+
+import java.util.Map;
+
+/**
+ * Elasticsearch plugins may provide an implementation of this class (via SPI) in order to add extra fields to the JSON based log file.
+ *
+ * @see DynamicContextDataProvider
+ */
+public interface LoggingDataProvider {
```

Comment:
I wonder if we want to expose this via libs:logging instead (which is kind of a stable plugin logging api) ?
this would allow plugins to implement the 'prefix logger' style of feature, so one less thing that we would have to support at some point in the future
wdyt @rjernst ?

Reply:
We may want to do that, but I suggest discussing it as a followup.